jenkins/Makefile

```diff
@@ -1,13 +1,68 @@
+#
 # A simple Makefile for building and pushing the various pbench-devel-*
-# container images.
+# and pbench-rpmbuild container images.
+#
+# This makefile defines the following targets:
 #
-# WARNING: Do not publish images layered on top of UBI images.
+#  all (the default):  builds the CI image and a Fedora-based development image
+#  image-ci:           builds the CI image
+#  image-fedora:       builds the Fedora-based development image
+#  image-rpmbuild-all: builds all the default containers for building RPMs
+#  image-rpmbuild-rhel-7:  builds a single RPM build container (e.g., for RHEL-7)
+#  push-ci:            pushes the CI image to the registry
+#  push-fedora:        pushes the Fedora-based development image to the registry
+#  push-rpmbuild-all:  pushes all the default RPM build containers to the registry
+#
+
+# Include definition of _DISTROS
+include ../utils/utils.mk
+
+# Default list of distributions for which to build containers in which to build RPMs
+RPMBUILD_BASEIMAGE_DEFAULTS = \
+	rhel-9 rhel-8 rhel-7 \
+	fedora-36 fedora-35 fedora-34 \
+	centos-9 centos-8
+
+# Registry where the resulting RPM build container images are to be stored
+RPMBUILD_IMAGE_REPO = images.paas.redhat.com/pbench
+
+# Templates for the various distributions' base images.  For the moment, for a
+# given distribution, all of the versions can be found by tweaking the image
+# tag; so, to generate the appropriate repository for the base image, we just
+# have to pick the appropriate template and insert the requested version, which
+# is exactly what happens when RPMBUILD_BASEIMAGE_REPO is expanded.
 #
+# Note that registry.access.redhat.com is moving to registry.redhat.io where it
+# will no longer permit anonymous access, but it works for the moment.
+VER := __version__
+RPMBUILD_BASEIMAGE_rhel = registry.access.redhat.com/ubi${VER}:latest
+RPMBUILD_BASEIMAGE_fedora = quay.io/fedora/fedora:${VER}
+RPMBUILD_BASEIMAGE_centos = quay.io/centos/centos:stream${VER}
+RPMBUILD_BASEIMAGE_REPO = $(subst ${VER},${DIST_VERSION},${RPMBUILD_BASEIMAGE_${DIST_NAME}})
+
+# List of packages required to build RPMs, to be installed in the build
+# containers.  (The jinja2 CLI is not available as an RPM on RHEL, so we
```

Comment:
This seems oddly specific and yet evasive. The reference to jinja2 cli is intended to implicitly justify the inclusion of `python3-pip` but you avoid actually saying that, which could leave a future reader scratching their head a moment. That is, "we install [jinja2 cli] via pip" is not what's happening here: we're installing `python3-pip` here so that we can later use it to install jinja2 cli.

Reply:
Actually, you misunderstand (and, if you can suggest how to fix the comment, I'd be obliged):  we _would_ install the `python3-jinja2-cli` RPM package, but that's available only on Fedora and CentOS, not on RHEL.  So, rather than implement a bunch of conditional code, we omit it from this list and install it via Pip (unconditionally), later (at line 86).

As for the installation of `python3-pip`, we use it inside the build of the RPM (at least for the Agent), to "install" our Python-based tools, and so we need it regardless of how we address the Jinja2 requirement.

Reply:
OK, fine, that's not why you put in the phrase; but that makes it even more out of place. Just drop the entire jinja2 cli comment. It's got nothing to do with the code here at all.